ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -615,6 +599,13 @@ public StatsSnapshot queryAggregatedPartitionClassStatsByClusterName(String clus
     return new StatsSnapshot(partitionClassNameValue, partitionClassNameSubMap);
   }
 
+  @Override
+  public void shutdown() {
+    if (dataSource instanceof HikariDataSource) {
```

Comment:
could this just be `instanceof Closeable` or are there some datasources we don't want to close?